hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/SolvencyPreCheck.java

```diff
@@ -144,13 +160,13 @@ public void checkSolvency(
         final var availableBalance = account.tinybarBalance();
         final var offeredFee = txBody.transactionFee();
 
-        if (offeredFee < fees.networkFee()) {
+        if (checkOfferedFee && offeredFee < fees.networkFee()) {
             throw new InsufficientNetworkFeeException(INSUFFICIENT_TX_FEE, totalFee);
         }
         if (availableBalance < fees.networkFee()) {
             throw new InsufficientNetworkFeeException(INSUFFICIENT_PAYER_BALANCE, totalFee);
         }
-        if (offeredFee < totalFee) {
+        if (checkOfferedFee && offeredFee < totalFee) {
```

Comment:
Why would we ever _not_ check the offered fee?

Reply:
This doesn't happen for child transactions. Only for USER and SCHEDULED